### PR TITLE
Pageviews

### PR DESCRIPTION
**Pull request by jamesgorrie**
Adds page tracking to `/works` and `/work`.

What we are heading towards would be a session that might look like:

| action | url | analytics event |
|--------|-----|------------------|
| visit | `/collections` | `pageview`
| search for gnomes in images | `/images?query=gnomes` | `pageview`
| click result 3 | `/images?query=gnomes` | `image search result selected`
| goes to page | `/works/123/images?id=abcdef` | `pageview`
| click work link on image page | `/works/123/images?id=abcdef` | `untracked`
| land on the work page | `/works/123` | `pageview`

__also__
* Updates works => TS
* renamed worksresults components to be more specific to what they are
* created a `WorkLink`. I'll continue on this patter and slowly phase out some of the complexity from routes, which you can do more neatly in TS.
* tidied up the error handling on the `work` page, more to do, but the PR was getting large

There is a mega annoyance here that in Segment our page event is currently blocked for some or other reason. If this proved to be annoying, we might just revert to using the [Track event](https://segment.com/docs/connections/spec/track/), which is actually less verbose in what it collects.

### CODE REVIEW

File: catalogue/webapp/pages/work.tsx [hunk 1]
```diff
@@ -1,40 +1,29 @@
...
-          title={workResponse.description}
-          globalContextData={globalContextData}
-        />
-      )}
-      {workResponse.type !== 'Error' && (
-        <Work work={workResponse} globalContextData={globalContextData} />
-      )}
-    </>
-  );
+  // TODO: move the <Work> ocmponent content back into here
```
Comment (gestchild) [inline catalogue/webapp/pages/work.tsx:25]: can get rid of this comment

File: catalogue/webapp/pages/works.tsx [hunk 2]
```diff
@@ -355,16 +340,29 @@ const Works = ({
...
 
-export const getServerSideProps = async (
-  ctx: Context
-): Promise<{ props: Props }> => {
-  const globalContextData = getGlobalContextData(ctx);
-  const params = WorksRoute.fromQuery(ctx.query);
-  const { enableColorFiltering } = ctx.query.toggles;
-  const _queryType = cookies(ctx)._queryType;
+export const t: GetServerSideProps<{ things: string }> = async context => {
+  console.info(context);
```
Comment (gestchild) [inline catalogue/webapp/pages/works.tsx:344]: left over console.log

File: catalogue/webapp/pages/work.tsx (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/work.tsx]: The `TODO` was added as I didn't want to clutter this PR any more. It wasn't very clear though, is [this](https://github.com/wellcomecollection/wellcomecollection.org/pull/5819/commits/ff4586a1d9addfed46fce17f25de8c407bf24c92) better?

File: catalogue/webapp/pages/works.tsx [hunk 3]
```diff
@@ -355,16 +340,29 @@ const Works = ({
...
+
+  const things = await Promise.resolve('things');
```
Comment (gestchild) [inline catalogue/webapp/pages/works.tsx:346]: need this anymore?

File: catalogue/webapp/pages/work.tsx (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/pages/work.tsx]: yep, misunderstood that

File: catalogue/webapp/pages/works.tsx (hunk 3 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/works.tsx]: removed the whole thing, was doing some typechecking debugging.

**Review by gestchild — APPROVED** [review]
looks good to me

File: catalogue/webapp/pages/works.tsx [hunk 4]
```diff
@@ -355,16 +340,20 @@ const Works = ({
...
-  const params = WorksRoute.fromQuery(ctx.query);
-  const { enableColorFiltering } = ctx.query.toggles;
-  const _queryType = cookies(ctx)._queryType;
+export const getServerSideProps: GetServerSideProps<
+  Props | AppErrorProps
+> = async context => {
+  const globalContextData = getGlobalContextData(context);
+  const params = WorksRoute.fromQuery(context.query);
+  // TODO: Fix this, we should have the toggles typesd at this point
+  const { enableColorFiltering } = (context.query.toggles as unknown) as Record<
```
Comment (ishouty) [inline catalogue/webapp/pages/works.tsx:349]: would be ideal if we had the type for the togglesContext

Comment (jamesgorrie) [reply catalogue/webapp/pages/works.tsx]: The trouble is propagating that time down the pipes.

As we've hacked it a little and attached the cached values onto `context.query`, which it self has a type of `{ [key: string]?: string | string[] }` as it's representing the querystring type from nodes http type system.

It's a part of the system I have really tried to work out... although now that we're speaking about it, [I think I know the answer for this case](https://github.com/wellcomecollection/wellcomecollection.org/pull/5819/commits/8713de1cbf8de1d8a81889eede4a952e63908550).

**Review by ishouty — APPROVED** [review]

File: common/services/conversion/track.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+declare global {
+  interface Window {
+    analytics: any;
+  }
+}
+
+type Page = {
+  path: string;
+  pathname: string;
+  name: string;
```
Comment (jamesgorrie) [inline common/services/conversion/track.ts:14]: constrain this list to only possible values

Comment (jamesgorrie) [reply common/services/conversion/track.ts:14]: Maybe add a description of this list in comments

**Pull request merged**